tests/optimagic/parameters/test_scaling.py

```diff
@@ -1,6 +1,33 @@
+from typing import get_args, get_type_hints
+
 import pytest
 from optimagic.exceptions import InvalidScalingError
-from optimagic.parameters.scaling import ScalingOptions, pre_process_scaling
+from optimagic.parameters.scaling import (
+    ScalingOptions,
+    ScalingOptionsDict,
+    pre_process_scaling,
+)
+
+
+def test_scaling_options_and_dict_have_same_attributes():
```

Comment:
Since this will occur in many places we should have one test file just dedicated to comparing TypedDicts and dataclasses and make the comparison function a bit more abstract.